### DISCUSSION TITLE
Get the selected option of the select field as an array

### DISCUSSION THREAD

**Discussion by ArtMin96** [filamentphp/filament#9640] [Help]
How can I get the selected option of the select field as an array: `value => name`.

```php
Select::make('from')
    ->options($this->getFromDestinationOption())
    ->suffixAction(
        Action::make('switch')
            ->icon('some-icon')
            ->action(function (Get $get, Set $set) {
               // Get from and to options as [key => value]
            })
    ),
Select::make('to')
    ->options($this->getToDestinationOption()),

**Comment by leandrocfe** [comment]
this?
```php
->action(function (\Livewire\Component $livewire, Component $component) {
    $formStatePath = $livewire->getFormStatePath();
    $from = $component->getContainer()->getComponent($formStatePath.'.from');
    $fromOptions = $from->getOptions();
})
```

**Reply by ArtMin96** [reply]
Thank you @leandrocfe. `getFormStatePath` returned null, so I tried to hardcode the `statePath`. But `getOptions` returned an empty array. I'm getting the options through the API, maybe that's the problem?

![image](https://github.com/filamentphp/filament/assets/29732308/d26568f5-d99c-4720-8da3-331c603a530b)

**Reply by ArtMin96** [reply]
I want to move selected options of selects one with another. And I use the form package.

**Reply by ArtMin96** [reply]
Yes, the problem is that the options are searched through the API.
